lib/disk.js

```diff
@@ -0,0 +1,67 @@
+const EventEmitter = require('events');
+const util = require('util');
+const fs = require('fs');
+
+function FileDisk(path) {
+	EventEmitter.call(this);
+	this.path = path;
+	this.fd = null;
+
+	this.open();
+}
+util.inherits(FileDisk, EventEmitter);
+
+FileDisk.prototype.open = function() {
+	let self = this;
+	// FIXME: allow opening the file readonly
+	fs.open(this.path, 'r+', function(err, fd) {
+		if (err) {
+			self.fd = -1;
+			return;
+		}
+		self.fd = fd;
+		self.emit('open')
+	});
+};
+
+FileDisk.prototype.request = function(type, offset, iovecs, callback) {
+	if (typeof this.fd !== 'number') {
+		return this.once('open', function() {
+			this.request(type, offset, iovecs, callback);
+		});
+	}
+
+	switch (type) {
+		// FIXME: expose human readable constants like LKL_DEV_BLK_READ etc
+		case 0:
+			// FIXME: we have to fill all the iovecs, not just the first
```

Comment:
Yes, I expect it to be very common actually. The kernel uses scatter/gather I/O which allows it to do DMA without requiring big contiguous buffers in memory. So with enough memory fragmentation a particular block request will need do be split into multiple iovecs.

That said, since not all disk backends directly benefit from this zero-copy method fulfiling block requests I think we should provide wrapper methods that expose a contiguous buffer interface and do the split/merge of iovecs in the background, similar to how there are `read` and `readv` system calls